### DISCUSSION TITLE
Signup issues on prod

### DISCUSSION THREAD

**Discussion by arpitdalal** [epicweb-dev/epic-stack#474] [Q&A] [answered]
I've deployed the app on fly following the steps mentioned in the [deployment documentation](https://github.com/epicweb-dev/epic-stack/blob/main/docs/deployment.md). But signing up gives me this error in the logs:
```
prisma:error
Invalid `prisma.session.create()` invocation:
An operation failed because it depends on one or more records that were required but not found. Expected 1 records to be connected, found only 0.
PrismaClientKnownRequestError:
Invalid `prisma.session.create()` invocation:
An operation failed because it depends on one or more records that were required but not found. Expected 1 records to be connected, found only 0.
at wn.handleRequestError (/myapp/node_modules/@prisma/client/runtime/library.js:123:6730)
at wn.handleAndLogRequestError (/myapp/node_modules/@prisma/client/runtime/library.js:123:6119)
at wn.request (/myapp/node_modules/@prisma/client/runtime/library.js:123:5839)
at l (/myapp/node_modules/@prisma/client/runtime/library.js:128:9763)
at signup (file:///myapp/app/utils/auth.server.ts:130:18)
at Object.transform (file:///myapp/app/routes/_auth+/onboarding.tsx:85:21)
at ZodEffects.safeParseAsync (file:///myapp/node_modules/zod/lib/index.mjs:677:24)
at action5 (file:///myapp/app/routes/_auth+/onboarding.tsx:67:21)
at /myapp/node_modules/@sentry/src/utils/instrumentServer.ts:207:1
at Object.callRouteActionRR (/myapp/node_modules/@remix-run/server-runtime/dist/data.js:35:16) {
  code: 'P2025',
  clientVersion: '5.3.1',
  meta: { cause: 'Expected 1 records to be connected, found only 0.' }
}
```

This error is logged after hitting submit on the onboarding page but this playwright test did pass in GH actions but this isn't working on prod for some reason 🤷‍♂️
I believe it's getting that error when trying to connect roles in the signup function [here](https://github.com/epicweb-dev/epic-stack/blob/b48be8d63f1c3d3b9f88a4c891db6467304aac18/app/utils/auth.server.ts#L133) but there are no roles yet so it cannot connect. Now I haven't created any roles in the DB myself. Do the deployment steps include creating these roles? If not, how do I create them?

**Comment by arpitdalal** [comment]
playwright tests work because the DB is seeded

**Comment by kiliman** [comment, marked as answer]
Did you follow the instructions from the "Seeding Production" section of the Deployment documents?

https://github.com/epicweb-dev/epic-stack/blob/main/docs/deployment.md#seeding-production

By default, the Dockerfile will generate a blank database. You'll need to either seed it with `sqlite3` cli, or you can update the `seed.ts` file to handle _production_ seeding. Make sure your script checks to see if the database has already been seeded. Update your Dockerfile:

```bash
ADD prisma .
RUN npx prisma generate
RUN npx prisma seed
```

You can either use `process.env.NODE_ENV` or pass values to your script:
https://www.prisma.io/docs/guides/migrate/seed-database#user-defined-arguments

**Reply by kentcdodds** [reply]
We should probably make a script to make this easier due to the fact that this is a requirement of the stack now.

**Reply by arpitdalal** [reply]
seed.ts already does this by setting MINIMAL_SEED env variable, right? Just need to run it when deploying for the first time.

**Reply by kentcdodds** [reply]
Replied in https://github.com/epicweb-dev/epic-stack/issues/475
